addons/mail/static/tests/owl/component/activity_tests.js

```diff
@@ -0,0 +1,832 @@
+        activity.querySelectorAll('.o_Activity_sidebar').length,
+        1,
+        "should have activity sidebar in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_mailTemplates').length,
+        1,
+        "should have activity mail templates in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_mailTemplate').length,
+        1,
+        "should have activity mail template in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_MailTemplate_name').length,
+        1,
+        "should have activity mail template name in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelector('.o_MailTemplate_name').textContent,
+        "Dummy mail template",
+        "should have activity mail template name in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_MailTemplate_preview').length,
+        1,
+        "should have activity mail template name preview button in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_MailTemplate_send').length,
+        1,
+        "should have activity mail template name send button in DOM"
+    );
+
+
+});
+
+QUnit.test('activity with mail template: preview mail', async function (assert) {
+    assert.expect(3);
+
+    await this.start({
+        intercepts: {
+            do_action: function (event) {
+                assert.deepEqual(
+                    event.data.action,
+                    {
+                        context: {
+                            default_res_id: 42,
+                            default_model: 'res.partner',
+                            default_use_template: true,
+                            default_template_id: 1,
+                            force_email: true,
+                        },
+                        name: 'Compose Email',
+                        type: 'ir.actions.act_window',
+                        res_model: 'mail.compose.message',
+                        views: [[false, 'form']],
+                        target: 'new',
+                    },
+                    "should open the preview mail form view"
+                );
+            },
+        },
+    });
+    const activityLocalId = this.env.store.dispatch('_createActivity', {
+        mail_template_ids: [{ id: 1, name: "Dummy mail template" }],
+        res_id: 42,
+        res_model: 'res.partner',
+    });
+    await this.createActivityComponent(activityLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Activity').length,
+        1,
+        "should have activity component in DOM"
+    );
+    const activity = document.querySelector('.o_Activity');
+    assert.strictEqual(
+        activity.querySelectorAll('.o_MailTemplate_preview').length,
+        1,
+        "should have activity mail template name preview button in DOM"
+    );
+
+    await activity.querySelector('.o_MailTemplate_preview').click();
```

Comment:
should have `assert.step()` and `assert.verifySteps()`, otherwise test fail with wrong assertion count if the intercept fails (and always assume other developers will simply react to symptoms rather than causes, so they will think "counter is too high? Let's decrement to make it pass!")

_(this comment applies to whole diff)_